<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1024</width>
    <height>768</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Code Designer</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_2">
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <item>
     <layout class="QHBoxLayout" name="hLayoutTools">
      <item>
       <widget class="QComboBox" name="cmbFunctions">
        <property name="minimumSize">
         <size>
          <width>160</width>
          <height>0</height>
         </size>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="btnAddFunction">
        <property name="toolTip">
         <string>Add Function</string>
        </property>
        <property name="icon">
         <iconset resource="../res/res.qrc">
          <normaloff>:/images/icon_add.png</normaloff>:/images/icon_add.png</iconset>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="btnDelFunction">
        <property name="toolTip">
         <string>Delete Function</string>
        </property>
        <property name="icon">
         <iconset resource="../res/res.qrc">
          <normaloff>:/images/icon_del.png</normaloff>:/images/icon_del.png</iconset>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QPushButton" name="btnGlobalVariables">
        <property name="text">
         <string>Global Variables</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="btnBinCodes">
        <property name="text">
         <string>Bin Codes</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="FlowView" name="flowView"/>
    </item>
   </layout>
  </widget>
  <widget class="DockWidget" name="dockWidgetFunctions">
   <property name="minimumSize">
    <size>
     <width>200</width>
     <height>40</height>
    </size>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="ToolBox" name="toolBox">
    <layout class="QVBoxLayout" name="verticalLayout"/>
   </widget>
  </widget>
  <widget class="DockWidget" name="dockWidgetProperties">
   <property name="minimumSize">
    <size>
     <width>200</width>
     <height>38</height>
    </size>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContentsProperties">
    <layout class="QVBoxLayout" name="verticalLayout_4">
     <property name="leftMargin">
      <number>0</number>
     </property>
     <property name="topMargin">
      <number>4</number>
     </property>
     <property name="rightMargin">
      <number>0</number>
     </property>
     <property name="bottomMargin">
      <number>0</number>
     </property>
     <item>
      <widget class="PropertyWidget" name="propertyWdiget" native="true"/>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="DockWidget" name="dockWidgetModules">
   <property name="minimumSize">
    <size>
     <width>200</width>
     <height>38</height>
    </size>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContentsModules"/>
  </widget>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionNew"/>
   <addaction name="actionOpen"/>
   <addaction name="actionSave"/>
   <addaction name="separator"/>
   <addaction name="actionAlignLeft"/>
   <addaction name="actionAlignHCenter"/>
   <addaction name="actionAlignRight"/>
   <addaction name="actionAlignTop"/>
   <addaction name="actionAlignVCenter"/>
   <addaction name="actionAlignBottom"/>
   <addaction name="separator"/>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1024</width>
     <height>23</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionNew"/>
    <addaction name="actionOpen"/>
    <addaction name="actionSave"/>
    <addaction name="actionSaveAs"/>
   </widget>
   <widget class="QMenu" name="menuTool">
    <property name="title">
     <string>Tool</string>
    </property>
    <addaction name="actionGlobalVariables"/>
    <addaction name="actionBinCodes"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuTool"/>
  </widget>
  <action name="actionNew">
   <property name="icon">
    <iconset resource="../res/res.qrc">
     <normaloff>:/images/icon_new.png</normaloff>:/images/icon_new.png</iconset>
   </property>
   <property name="text">
    <string>New</string>
   </property>
   <property name="toolTip">
    <string>New File</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="actionOpen">
   <property name="icon">
    <iconset resource="../res/res.qrc">
     <normaloff>:/images/icon_open.png</normaloff>:/images/icon_open.png</iconset>
   </property>
   <property name="text">
    <string>Open</string>
   </property>
   <property name="toolTip">
    <string>Open File</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="icon">
    <iconset resource="../res/res.qrc">
     <normaloff>:/images/icon_save.png</normaloff>:/images/icon_save.png</iconset>
   </property>
   <property name="text">
    <string>Save</string>
   </property>
   <property name="toolTip">
    <string>Save File</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionAlignLeft">
   <property name="icon">
    <iconset resource="../res/res.qrc">
     <normaloff>:/images/icon_align_left.png</normaloff>:/images/icon_align_left.png</iconset>
   </property>
   <property name="text">
    <string>Align Left</string>
   </property>
   <property name="toolTip">
    <string>Align Left</string>
   </property>
   <property name="shortcut">
    <string>Alt+L</string>
   </property>
  </action>
  <action name="actionAlignHCenter">
   <property name="icon">
    <iconset resource="../res/res.qrc">
     <normaloff>:/images/icon_align_hcenter.png</normaloff>:/images/icon_align_hcenter.png</iconset>
   </property>
   <property name="text">
    <string>Align Horizon Center</string>
   </property>
   <property name="toolTip">
    <string>Align Horizon Center</string>
   </property>
   <property name="shortcut">
    <string>Alt+H</string>
   </property>
  </action>
  <action name="actionAlignRight">
   <property name="icon">
    <iconset resource="../res/res.qrc">
     <normaloff>:/images/icon_align_right.png</normaloff>:/images/icon_align_right.png</iconset>
   </property>
   <property name="text">
    <string>Align Right</string>
   </property>
   <property name="toolTip">
    <string>Align Horizon Center</string>
   </property>
   <property name="shortcut">
    <string>Alt+R</string>
   </property>
  </action>
  <action name="actionAlignTop">
   <property name="icon">
    <iconset resource="../res/res.qrc">
     <normaloff>:/images/icon_align_top.png</normaloff>:/images/icon_align_top.png</iconset>
   </property>
   <property name="text">
    <string>Align Top</string>
   </property>
   <property name="toolTip">
    <string>Align Top</string>
   </property>
   <property name="shortcut">
    <string>Alt+T</string>
   </property>
  </action>
  <action name="actionAlignVCenter">
   <property name="icon">
    <iconset resource="../res/res.qrc">
     <normaloff>:/images/icon_align_vcenter.png</normaloff>:/images/icon_align_vcenter.png</iconset>
   </property>
   <property name="text">
    <string>Align Vertical Center</string>
   </property>
   <property name="toolTip">
    <string>Align Vertical Center</string>
   </property>
   <property name="shortcut">
    <string>Alt+V</string>
   </property>
  </action>
  <action name="actionAlignBottom">
   <property name="icon">
    <iconset resource="../res/res.qrc">
     <normaloff>:/images/icon_align_bottom.png</normaloff>:/images/icon_align_bottom.png</iconset>
   </property>
   <property name="text">
    <string>Align Bottom</string>
   </property>
   <property name="toolTip">
    <string>Align Bottom</string>
   </property>
   <property name="shortcut">
    <string>Alt+B</string>
   </property>
  </action>
  <action name="actionGlobalVariables">
   <property name="text">
    <string>Global Variables</string>
   </property>
  </action>
  <action name="actionBinCodes">
   <property name="text">
    <string>Bin Codes</string>
   </property>
  </action>
  <action name="actionSaveAs">
   <property name="icon">
    <iconset resource="../res/res.qrc">
     <normaloff>:/images/icon_saveas.png</normaloff>:/images/icon_saveas.png</iconset>
   </property>
   <property name="text">
    <string>Save As</string>
   </property>
   <property name="toolTip">
    <string>Save As</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+S</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ToolBox</class>
   <extends>QWidget</extends>
   <header>ui/ToolBox.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>DockWidget</class>
   <extends>QDockWidget</extends>
   <header>ui/DockWidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>FlowView</class>
   <extends>QGraphicsView</extends>
   <header>ui/FlowView.h</header>
  </customwidget>
  <customwidget>
   <class>PropertyWidget</class>
   <extends>QWidget</extends>
   <header>ui/PropertyWidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../res/res.qrc"/>
 </resources>
 <connections/>
</ui>
